### PR TITLE
Storage API with MinIO support

### PR DESCRIPTION
**Pull request by mocsharp**
- Migrated storage layer source code from Informatics Gateway
- Setup CI to:
  - build
  - publish all builds to GitHub packages
  - publish `main` and `release/*` builds to NuGet
  - Perform CodeQL Analysis
  - Secret detection
  - Run unit test if any
  - SonarCloud report
  - CodeCov report

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (evberrypi) [inline .github/ISSUE_TEMPLATE/feature_request.md:1]: SPDX?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+---
```
Comment (evberrypi) [inline .github/ISSUE_TEMPLATE/bug_report.md:1]: SPDX?

File: src/.sonarlint/Monai.Deploy.Storage.slconfig [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "ServerUri": "https://sonarcloud.io/",
+  "Organization": {
+    "Key": "project-monai",
+    "Name": "Project MONAI"
+  },
+  "ProjectKey": "Project-MONAI_monai-deploy-storage",
+  "ProjectName": "monai-deploy-storage",
+  "Profiles": {
+    "CSharp": {
+      "ProfileKey": "AX96ONQSnTk2GEVJ9ILJ",
```
Comment (evberrypi) [inline src/.sonarlint/Monai.Deploy.Storage.slconfig:11]: Should this key be present?

**Review by evberrypi — COMMENTED** [review]
Signing off on CI/workflow changes (not C#)

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 1 above)
Comment (mocsharp) [reply .github/ISSUE_TEMPLATE/feature_request.md:1]: Not sure if we want to include SPDX in an issue template.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (mocsharp) [reply .github/ISSUE_TEMPLATE/bug_report.md:1]: Not sure if we want to include SPDX in an issue template.

File: src/.sonarlint/Monai.Deploy.Storage.slconfig (hunk 3 above)
Comment (mocsharp) [reply src/.sonarlint/Monai.Deploy.Storage.slconfig:11]: Yes, this is not a secret but a reference to a sonar language profile.

**Pull request merged**